monetio/sat/_tropomi_l2_no2_mm.py

```diff
@@ -103,16 +184,14 @@ def apply_quality_flag(ds):
     ds : xarray.Dataset
     """
     if "quality_flag" in ds.attrs:
-        quality_flag = ds[ds.attrs["quality_flag"]]
+        quality_flag = ds[ds.attrs["quality_flag"]]  # quality flag variable (float)
         quality_thresh_min = ds.attrs["quality_thresh_min"]
 
-        # Apply the quality thresh minimum to all variables in ds
-        for varname in ds:
-            print(varname)
-            if varname != ds.attrs["quality_flag"]:
-                logging.debug(varname)
-                values = ds[varname].values
-                values[quality_flag <= quality_thresh_min] = np.nan
+        # Apply the quality thresh minimum to selected variables
+        for varname in ds.attrs["var_applied"]:
+            logging.debug(f"applying quality flag to {varname}")
+            values = ds[varname].values
+            values[quality_flag <= quality_thresh_min] = np.nan
 
 
 def open_dataset(fnames, variable_dict, debug=False):
```

Comment:
can we change the function name to **read_trpdataset**? Just to be consistent with driver which has been incorporated into melodies-monet

Reply:
According to the pyYaml docs (https://pyyaml.org/wiki/PyYAMLDocumentation), None should be represented either with ~ or with null

Reply:
I added it as an alias: 626b95c 

In general, we're trying to keep API's for the different model/sat readers somewhat consistent, with the same function names etc.